src/connect4.ts

```diff
@@ -85,7 +90,7 @@ export function playToken(
   }
 
   boardState[(index > -1 ? index : boardLayout.NB_ROWS) - 1][column - 1] = pNum;
-  return boardState;
+  return { ...boardState };
```

Comment:
Wrong fix: you've already modified the argument in the previous line. Instead, start by cloning the argument (using [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/Window/structuredClone#syntax)), and modify the clone.